# Disallow multiple access of same DOM element (`@ecocode/increment-check`)


## Rule details



## Examples

Examples of **incorrect** code for this rule:

```js
test
```

Examples of **correct** code for this rule:

```js

```